subprojects/language-groovy/src/main/java/org/gradle/api/tasks/javadoc/Groovydoc.java

```diff
@@ -344,18 +354,98 @@ public void setOverviewText(@Nullable TextResource overviewText) {
     }
 
     /**
-     * Returns whether to include all classes and members (i.e. including private ones).
+     * Returns whether to include classes with private access and above.
+     *
+     * @deprecated Equivalent to calling {@link #getAccess()}{@code .get()} and checking for equivalence with {@link GroovydocAccess#PRIVATE}
      */
-    @Input
+    @Internal
+    @Deprecated
     public boolean isIncludePrivate() {
-        return includePrivate;
+        DeprecationLogger.deprecateMethod(Groovydoc.class, "isIncludePrivate()")
+            .replaceWith("getAccess")
+            .willBeRemovedInGradle8()
+            .withUpgradeGuideSection(7, "groovydoc_option_improvements")
+            .nagUser();
+        return getAccess().get() == GroovydocAccess.PRIVATE;
     }
 
     /**
-     * Sets whether to include all classes and members (i.e. including private ones) if set to true.
+     * Sets whether to include classes and members with private access and above.
+     *
+     * @deprecated Equivalent to calling {@link #setAccess(String)} with {@code "private"}
+     *             if {@code includePrivate} is {@code true}, {@code "public"} otherwise
      */
+    @Deprecated
     public void setIncludePrivate(boolean includePrivate) {
-        this.includePrivate = includePrivate;
+        DeprecationLogger.deprecateMethod(Groovydoc.class, "setIncludePrivate(boolean)")
+            .replaceWith("getAccess")
+            .willBeRemovedInGradle8()
+            .withUpgradeGuideSection(7, "groovydoc_option_improvements")
+            .nagUser();
+        getAccess().set(includePrivate ? GroovydocAccess.PRIVATE : GroovydocAccess.PUBLIC);
+    }
+
+    /**
+     * The smallest access level to include in the Groovydoc.
+     *
+     * <p>
+     * For example, to include classes and members with package, protected, and public access, use {@link GroovydocAccess#PACKAGE}.
+     * </p>
+     *
+     * @return the access property
+     * @since 7.5
+     */
+    @Input
+    public Property<GroovydocAccess> getAccess() {
+        return access;
+    }
+
+    /**
+     * Sets the smallest access level to include in the Groovydoc.
+     *
+     * <p>
+     * For example, to include classes and members with package, protected, and public access, use {@code "package"}.
+     * </p>
+     *
+     * <p>
+     * Equivalent to {@code getAccess().set(GroovydocAccess.valueOf(access.toUpperCase(Locale.ROOT)))}.
+     * </p>
+     *
+     * @param access the smallest access to include
+     * @since 7.5
+     */
+    public void setAccess(String access) {
+        getAccess().set(GroovydocAccess.valueOf(access.toUpperCase(Locale.ROOT)));
+    }
+
```

Comment:
I don't think we want to add this since everything should go through the property.

There are two options:
1. Require users to use `GroovydocAccess.PRIVATE` here
2. Support strings for all `Property<Enum>` types

(2) looks something like this: https://github.com/gradle/gradle/commit/429ad9ec53ab2e4c84835437cadc3659ea601ce8

Reply:
Removing this for now, will make (2) its own PR.